e2e/test-deployment.bats

```diff
@@ -9,8 +9,11 @@ DETIK_CLIENT_NAMESPACE="k8up-system"
 DEBUG_DETIK="true"
 
 @test "verify the deployment" {
-	apply definitions/deployment
-	echo "$output"
+	# Remove traces of k8up deployments from other tests
+	kubectl delete namespace "$DETIK_CLIENT_NAMESPACE" || true
+	kubectl create namespace "$DETIK_CLIENT_NAMESPACE" || true
+
+	apply definitions/k8up
```

Comment:
I would just delete this file. It was there as the first skeleton test since we didn't any e2e tests back then.
Besides, the operator is already transparently tested with this backup test now

Reply:
I like that it explicitly tests the deployment. When the other test fails, and it's because of the deployment, then it will be hard to interpret that. For example, because there are no `try` statements regarding the deployment, as these only apply to the current `DETIC_CLIENT_NAMESPACE`. (And this namespace is different in other tests, as we operator in `k8up-e2e-subject` there and not `k8up-system` as here.)

Reply:
Fine. I would prefer though a different file name. So that Bats takes the deployment test first instead of backup. Bats seems to choose the test files alphabetically by filename. Maybe like `00-test-deployment.bats`

Maybe we can use also another namespace for this? Or is that too much to configure?